src/components/DarkMode.js

```diff
@@ -0,0 +1,75 @@
+import React, { useEffect } from "react";
+import "./DarkMode.css";
+
+const DarkMode = () => {
+  // function to set dark theme
+  const setDark = () => {
+    localStorage.setItem("theme", "dark");
+
+    document.documentElement.setAttribute("data-theme", "dark");
+  };
+  // function to set light theme
+  const setLight = () => {
+    localStorage.setItem("theme", "light");
+    document.documentElement.setAttribute("data-theme", "light");
+  };
+
+  useEffect(() => {
+    const storedTheme = localStorage.getItem("theme");
+
+    const prefersDark =
+      window.matchMedia &&
+      window.matchMedia("(prefers-color-scheme: dark)").matches;
+
+    if (storedTheme === "dark" || (storedTheme === null && prefersDark)) {
+      setDark();
+    } else {
+      setLight();
+    }
+
+    // listen for changes in system preferences and update theme if no localStorage value
+    const mediaQuery = window.matchMedia("(prefers-color-scheme: dark)");
+    const handleChange = (e) => {
+      if (!localStorage.getItem("theme")) {
+        // mediaQuery.matches ? setDark() : setLight();
```

Comment:
Since this is commented out, you could remove this line.

Reply:
I removed this, and updated the code.